static/js/cm-editor.ts

```diff
@@ -117,7 +121,17 @@ export class HedyCodeMirrorEditor implements HedyEditor {
                     ...defaultKeymap,
                     ...searchKeymap, // we need to replace this with our own search widget
                     ...historyKeymap,
+                    indentWithTab,
                 ]),
+                indentUnit.of(indentSize),
+                indentService.of((context, pos) => {
+                    const previousLine = context.lineAt(pos, -1);
+                    const nextLine = context.lineAt(pos + 1, -1);
+                    const nextIndentationSize =  nextLine.text.match(/^\s*/)![0].length;
+                    const prevIndentationSize =  previousLine.text.match(/^\s*/)![0].length;
+                    const indentBy = Math.max(prevIndentationSize, nextIndentationSize);
+                    return indentBy
+                }),
```

Comment:
When there's no next line, this returns an error. Also, can you add this code to its own function?

Reply:
Good one, fixed!